[dogfood] SessionStart build hook doesn't prevent PreToolUse bootstrap failures in fresh worktrees

## Governance Issue

**Agent:** workspace-pr-review-agent (claude-code:unknown:reviewer)
**Version:** agentguard v2.10.2
**Repo:** AgentGuardHQ/agentguard-workspace (worktree: workspace-pr-review-agent-2706085)
**Box:** $(hostname) / $(whoami)
**Action attempted:** First Bash command after session start (reading agent registry)
**Expected behavior:** SessionStart hooks should have built `apps/cli/dist/bin.js` before any PreToolUse hook runs, ensuring governance is active from the first command.
**Actual behavior:** PreToolUse hook blocked ALL non-bootstrap Bash commands with _"AgentGuard kernel binary not found"_ despite the SessionStart hook being configured to build the binary. Manual `pnpm build` (bootstrap-safe command) was required to unblock.

### Root Cause Hypothesis

The `settings.json` SessionStart hook runs:
```
test -f apps/cli/dist/bin.js || pnpm build
```

When `pnpm build` runs via Turbo in a fresh worktree, Turbo uses a shared content-addressed cache. The build logs showed the cached `@red-codes/agentguard` package resolving to a **different worktree's path**:

```
@red-codes/agentguard:build: cache hit, replaying logs bbe6d350f5c87dc0
@red-codes/agentguard:build: > @red-codes/agentguard@2.10.2 build
  /home/jared/agentguard-workspace/.worktrees/pr-merger-agent-2104481/apps/cli
```

If Turbo extracted cache outputs to the ORIGINAL cached path (another worktree) rather than the CURRENT worktree, then `apps/cli/dist/bin.js` in the current worktree would remain absent — causing the PreToolUse hook to fail-closed.

Manual `pnpm build` from the bootstrap-safe exemption succeeded in 125ms (full cache hit) and produced the binary at the correct worktree path, unblocking governance.

### Observed Sequence

1. Session started in fresh worktree — binary absent
2. SessionStart hook `test -f apps/cli/dist/bin.js || pnpm build` ran (session-persona-check.sh ran immediately after, confirming hook sequence executed)
3. First Bash command → PreToolUse hook → **BLOCKED**: binary not found
4. Manual `pnpm build` → 125ms Turbo cache hit → binary created at worktree path
5. Governance became active; all subsequent Bash commands allowed

### Impact

Fresh worktrees created by the scheduler will consistently fail to self-bootstrap if pnpm/Turbo resolves cache outputs to the original worktree rather than the current one. This forces agents to know the bootstrap-safe escape hatch (`pnpm build`), which is not guaranteed for all agent SKILL.md definitions.

### Policy (agentguard.yaml)
<details><summary>agentguard.yaml</summary>

```yaml
$(cat agentguard.yaml 2>/dev/null | head -60 || echo '(not found)')
```
</details>

### Persona
```
AGENTGUARD_PERSONA_DRIVER=claude-code
AGENTGUARD_PERSONA_MODEL=unknown
AGENTGUARD_PERSONA_ROLE=reviewer
AGENTGUARD_PERSONA_PROJECT=workspace-pr-review-agent-2706085
```

### Hook settings
<details><summary>.claude/settings.json (SessionStart hooks)</summary>

```json
{
  "test -d node_modules || pnpm install --frozen-lockfile": "120s blocking",
  "test -f apps/cli/dist/bin.js || pnpm build": "120s blocking",
  "bash scripts/session-persona-check.sh": "5s blocking"
}
```
</details>

### Relevant log output
```
SessionStart hook success: [AgentGuard] No agent identity set for this session.
Project: workspace-pr-review-agent-2706085 | Model: unknown

[First Bash command]
PreToolUse:Bash hook blocking error: "bash scripts/claude-hook-wrapper.sh":
AgentGuard kernel binary not found — governance cannot evaluate this action.
Run: pnpm install && pnpm build
```

### Suggested Fix

Option A: In the SessionStart build hook, verify the binary exists AT THE CURRENT WORKTREE PATH after build:
```bash
pnpm build && test -f apps/cli/dist/bin.js || { echo "BUILD FAILED: binary not at expected path"; exit 1; }
```

Option B: Add a fallback direct build step that bypasses Turbo workspace resolution:
```bash
cd apps/cli && npx tsc -b && npx tsx esbuild.config.ts
```

Option C: Add an explicit Turbo `--output-logs=full` flag and verify cache extraction destinations in worktree contexts.

---
*Filed by workspace-pr-review-agent during scheduled run 2026-03-29*

--- a/.claude/settings.json
+++ b/.claude/settings.json
@@ -32,13 +32,13 @@
           },
           {
             "type": "command",
-            "command": "test -d node_modules || pnpm install --frozen-lockfile",
+            "command": "test -f node_modules/.modules.yaml || pnpm install --force",
             "timeout": 120000,
             "blocking": true
           },
           {
             "type": "command",
-            "command": "test -f apps/cli/dist/bin.js || pnpm build",
+            "command": "test -f apps/cli/dist/bin.js || pnpm build; test -f apps/cli/dist/bin.js",
             "timeout": 120000,
             "blocking": true
           },

--- a/apps/cli/src/commands/claude-init.ts
+++ b/apps/cli/src/commands/claude-init.ts
@@ -299,10 +299,22 @@ export async function claudeInit(args: string[] = []): Promise<void> {
     blocking: boolean;
   }> = [];
   if (isLocal) {
-    // In the agentguard dev repo, auto-build if dist is missing
+    // In the agentguard dev repo, ensure deps are linked and CLI is built.
+    // Use .modules.yaml as the pnpm install sentinel — written only after both
+    // the content-addressed store AND the symlink phase succeed (#1341).
+    // --force avoids the interactive "remove & reinstall" prompt in non-TTY envs.
     sessionStartHooks.push({
       type: 'command',
-      command: `test -f apps/cli/dist/bin.js || pnpm build`,
+      command: `test -f node_modules/.modules.yaml || pnpm install --force`,
+      timeout: 120000,
+      blocking: true,
+    });
+    // Verify the binary actually landed in THIS worktree after build.
+    // Turbo cache hits can replay to a different worktree path; the trailing
+    // `; test -f` turns that silent pass into a visible failure (#1322).
+    sessionStartHooks.push({
+      type: 'command',
+      command: `test -f apps/cli/dist/bin.js || pnpm build; test -f apps/cli/dist/bin.js`,
       timeout: 120000,
       blocking: true,
     });

--- a/apps/cli/tests/cli-claude-init.test.ts
+++ b/apps/cli/tests/cli-claude-init.test.ts
@@ -127,7 +127,7 @@ describe('claudeInit', () => {
     expect(written.hooks.SessionStart[0].hooks[1].command).toContain('status');
   });
 
-  it('installs SessionStart build + persona check + status hooks in local dev repo', async () => {
+  it('installs SessionStart install + build + persona check + status hooks in local dev repo', async () => {
     // Simulate being in the agentguard dev repo (apps/cli/src/bin.ts exists)
     vi.mocked(existsSync).mockImplementation((p: unknown) => {
       const path = String(p);
@@ -139,16 +139,22 @@ describe('claudeInit', () => {
 
     const written = writtenSettings();
     expect(written.hooks.SessionStart).toHaveLength(1);
-    expect(written.hooks.SessionStart[0].hooks).toHaveLength(3);
-    // Build hook first
-    expect(written.hooks.SessionStart[0].hooks[0].command).toContain('pnpm build');
+    expect(written.hooks.SessionStart[0].hooks).toHaveLength(4);
+    // Install check first — uses .modules.yaml sentinel (#1341)
+    expect(written.hooks.SessionStart[0].hooks[0].command).toContain('.modules.yaml');
+    expect(written.hooks.SessionStart[0].hooks[0].command).toContain('pnpm install');
     expect(written.hooks.SessionStart[0].hooks[0].blocking).toBe(true);
     expect(written.hooks.SessionStart[0].hooks[0].timeout).toBe(120000);
-    // Persona check second
-    expect(written.hooks.SessionStart[0].hooks[1].command).toContain('session-persona-check');
-    // Status hook third — in dev repo, uses local binary directly (no bash wrapper)
-    expect(written.hooks.SessionStart[0].hooks[2].command).toContain('status');
-    expect(written.hooks.SessionStart[0].hooks[2].command).toContain('apps/cli/dist/bin.js');
+    // Build hook second — trailing test verifies binary landed here (#1322)
+    expect(written.hooks.SessionStart[0].hooks[1].command).toContain('pnpm build');
+    expect(written.hooks.SessionStart[0].hooks[1].command).toContain('apps/cli/dist/bin.js');
+    expect(written.hooks.SessionStart[0].hooks[1].blocking).toBe(true);
+    expect(written.hooks.SessionStart[0].hooks[1].timeout).toBe(120000);
+    // Persona check third
+    expect(written.hooks.SessionStart[0].hooks[2].command).toContain('session-persona-check');
+    // Status hook fourth — in dev repo, uses local binary directly (no bash wrapper)
+    expect(written.hooks.SessionStart[0].hooks[3].command).toContain('status');
+    expect(written.hooks.SessionStart[0].hooks[3].command).toContain('apps/cli/dist/bin.js');
   });
 
   it('detects already-configured hook in PreToolUse and warns', async () => {

